WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/AddEditCouponViewModel.swift

```diff
@@ -396,6 +458,19 @@ private extension AddEditCouponViewModel {
         return coupon.freeShipping != initialCoupon.freeShipping
     }
 
+    func trackCouponCreateInitiated(with coupon: Coupon) {
+        ServiceLocator.analytics.track(.couponCreationInitiated, withProperties: [
+            "discount_type": coupon.discountType.rawValue,
+            "has_expiry_date": coupon.dateExpires != nil,
+            "includes_free_shipping": coupon.freeShipping,
+            "has_description": coupon.description.isNotEmpty,
+            "has_product_or_category_restrictions": coupon.excludedProductCategories.isNotEmpty || coupon.excludedProductIds.isNotEmpty,
+            "has_usage_restrictions": (coupon.usageLimit ?? 0) > 0
+            || (coupon.limitUsageToXItems ?? 0) > 0
+            || (coupon.usageLimitPerUser ?? 0) > 0
```

Comment:
For `has_usage_restrictions`, I'm thinking of something more complicated like what we did with [`checkUsageRestrictionsUpdated`](https://github.com/woocommerce/woocommerce-ios/blob/a99b74e251249f3a3e24a6e10ae4324f6b8cea56/WooCommerce/Classes/ViewRelated/Coupons/Add%20and%20Edit%20Coupons/AddEditCouponViewModel.swift#L332-L348) 😅 The idea is to cover all the restrictions defined there - does it make sense?

Reply:
Sorry for this sequence of mistakes, thanks for clarifying everything, it's fixed now here: https://github.com/woocommerce/woocommerce-ios/pull/7052/commits/69d7dc809915e50808c685a46795703c6be1b178